rich/console.py

```diff
@@ -1682,18 +1682,18 @@ def log(
             if style is not None:
                 renderables = [Styled(renderable, style) for renderable in renderables]
 
-            caller = inspect.stack()[_stack_offset]
-            link_path = (
-                None
-                if caller.filename.startswith("<")
-                else os.path.abspath(caller.filename)
-            )
-            path = caller.filename.rpartition(os.sep)[-1]
-            line_no = caller.lineno
+            caller = inspect.currentframe()
+            while _stack_offset and caller:
+                caller = caller.f_back
```

Comment:
Note that it might happen that `caller` becomes `None`. This would be the analog of a potential `IndexError` from the c current implementation.